src/mame/sinclair/sprinter.cpp

```diff
@@ -0,0 +1,1751 @@
+	                                                       PORT_CODE(KEYCODE_LALT) PORT_CODE(KEYCODE_RALT)
+	PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("m    M    .      PI       INVERSE  PAUSE") PORT_CODE(KEYCODE_M)        PORT_CHAR('m') PORT_CHAR('M') PORT_CHAR('.')
+	                                                                 PORT_CODE(KEYCODE_STOP)
+	PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("n    N    ,      INKEY$   OVER     NEXT") PORT_CODE(KEYCODE_N)         PORT_CHAR('n') PORT_CHAR('N') PORT_CHAR(',')
+	                                                                 PORT_CODE(KEYCODE_COMMA)
+	PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("b    B    *      BIN      BRIGHT   BORDER") PORT_CODE(KEYCODE_B)       PORT_CHAR('b') PORT_CHAR('B') PORT_CHAR('*')
+	                                                                 PORT_CODE(KEYCODE_ASTERISK)
+	PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("CS Line7") PORT_CODE(KEYCODE_ESC) PORT_CODE(KEYCODE_LALT) PORT_CODE(KEYCODE_RALT)
+	PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("SS Line7") PORT_CODE(KEYCODE_STOP) PORT_CODE(KEYCODE_COMMA) PORT_CODE(KEYCODE_ASTERISK)
+	PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_UNUSED)
+
+
+	PORT_START("mouse_input1")
+	PORT_BIT(0xff, 0, IPT_MOUSE_X) PORT_SENSITIVITY(30)
+
+	PORT_START("mouse_input2")
+	PORT_BIT(0xff, 0, IPT_MOUSE_Y) PORT_INVERT PORT_SENSITIVITY(30)
+
+	PORT_START("mouse_input3")
+	PORT_BIT(0xf8, IP_ACTIVE_LOW, IPT_UNUSED)
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_BUTTON4) PORT_NAME("Left mouse button") PORT_CODE(MOUSECODE_BUTTON1)
+	PORT_BIT(0x02, IP_ACTIVE_LOW, IPT_BUTTON5) PORT_NAME("Right mouse button") PORT_CODE(MOUSECODE_BUTTON2)
+	PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_BUTTON6) PORT_NAME("Middle mouse button") PORT_CODE(MOUSECODE_BUTTON3)
+
+
+	PORT_START("NMI")
+	PORT_BIT(0x01, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_NAME("NMI") PORT_CODE(KEYCODE_F11)
+
+	PORT_START("TURBO")
+	PORT_BIT(0x01, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_NAME("Turbo") PORT_CODE(KEYCODE_F12) PORT_TOGGLE PORT_CHANGED_MEMBER(DEVICE_SELF, sprinter_state, turbo_changed, 0)
+	PORT_BIT(0xfe, IP_ACTIVE_HIGH, IPT_UNUSED)
+INPUT_PORTS_END
+
+void sprinter_state::sprinter(machine_config &config)
+{
+	spectrum_128(config);
+	config.device_remove("palette");
+
+	m_ram->set_default_size("64M");
+
+	Z84C015(config.replace(), m_maincpu, X_SP / 12); // 3.5MHz default
+	m_maincpu->set_m1_map(&sprinter_state::map_fetch);
+	m_maincpu->set_memory_map(&sprinter_state::map_mem);
+	m_maincpu->set_io_map(&sprinter_state::map_io);
+	m_maincpu->nomreq_cb().set_nop();
+	m_maincpu->set_irq_acknowledge_callback(NAME([](device_t &, int){ return 0xff; }));
+	m_maincpu->irqack_cb().set(FUNC(sprinter_state::irq_off));
+
+	m_screen->set_raw(X_SP / 3, SPRINT_WIDTH, SPRINT_HEIGHT, { 0, SPRINT_XVIS - 1, 0, SPRINT_YVIS - 1 });
+	m_screen->set_screen_update(FUNC(sprinter_state::screen_update));
+
+	PALETTE(config, "palette", FUNC(sprinter_state::sprinter_palette), 256 * 8);
```

Comment:
Since this initialises the palette with black, can you just do it as:
```c++
 	PALETTE(config, "palette", palette_device::BLACK).set_entries(256 * 8);
```
and avoid the extra function?